arb_os/retrybuffer.mini

```diff
@@ -0,0 +1,357 @@
+//
+// Copyright 2021, Offchain Labs, Inc. All rights reserved.
+//
+
+use accounts::getGlobalAccountStore;
+use accounts::setGlobalAccountStore;
+use accounts::accountStore_get;
+use accounts::accountStore_set;
+use accounts::account_addToEthBalance;
+use accounts::account_deductFromEthBalance;
+
+use evmCallStack::evmCallStack_callRetryableTxAndReturn;
+
+use evmlogs::EvmLogs;
+
+use gasAccounting::GasUsage;
+use gasAccounting::txFeeStats_zero;
+use gasAccounting::chargeForRetryableTx;
+use gasAccounting::getRetryableTxPrice;
+use gasAccounting::txFeeStats_zero;
+
+use inbox::IncomingRequest;
+use inbox::inbox_currentTimestamp;
+
+use messages::TxRequestData;
+
+use output::emitTxReceipt;
+
+use std::bytearray::ByteArray;
+use std::bytearray::bytearray_new;
+use std::bytearray::bytearray_size;
+use std::bytearray::bytearray_get256;
+use std::bytearray::bytearray_set256;
+use std::bytearray::bytearray_extract;
+
+use std::keccak::keccak256;
+
+use std::queue::Queue;
+use std::queue::queue_new;
+use std::queue::queue_put;
+use std::queue::queue_get;
+
+use core::kvs::Kvs;
+use core::kvs::builtin_kvsDelete;
+
+
+type RetryableTx = struct {
+    txId: uint,
```

Comment:
Added a comment explaining this.  Basically the RetryableTx needs a different ID than the message that submitted it, because ArbOS will need to issue separate receipts for the two.